docs/Compatibility.md

```diff
@@ -4,156 +4,259 @@ The Ruby Datadog Trace library is open source. See the [dd-trace-rb][1] GitHub r
-| Redis                      | `redis`                    | `>= 3.2`                 | `>= 3.2`                  | [Link][38]                          | [Link](https://github.com/redis/redis-rb)                                    |
-| Resque                     | `resque`                   | `>= 1.0`                 | `>= 1.0`                  | [Link][39]                          | [Link](https://github.com/resque/resque)                                     |
-| Rest Client                | `rest-client`              | `>= 1.8`                 | `>= 1.8`                  | [Link][40]                          | [Link](https://github.com/rest-client/rest-client)                           |
-| Roda                       | `roda`                     | `>= 2.1, <4`             | `>= 2.1, <4`              | [Link][41]                          | [Link](https://github.com/jeremyevans/roda)                                  |
-| Sequel                     | `sequel`                   | `>= 3.41`                | `>= 3.41`                 | [Link][42]                          | [Link](https://github.com/jeremyevans/sequel)                                |
-| Shoryuken                  | `shoryuken`                | `>= 3.2`                 | `>= 3.2`                  | [Link][43]                          | [Link](https://github.com/phstc/shoryuken)                                   |
-| Sidekiq                    | `sidekiq`                  | `>= 3.5.4`               | `>= 3.5.4`                | [Link][44]                          | [Link](https://github.com/mperham/sidekiq)                                   |
-| Sinatra                    | `sinatra`                  | `>= 1.4`                 | `>= 1.4`                  | [Link][45]                          | [Link](https://github.com/sinatra/sinatra)                                   |
-| Sneakers                   | `sneakers`                 | `>= 2.12.0`              | `>= 2.12.0`               | [Link][46]                          | [Link](https://github.com/jondot/sneakers)                                   |
-| Stripe                     | `stripe`                   | `>= 5.15.0`              | `>= 5.15.0`               | [Link][47]                          | [Link](https://github.com/stripe/stripe-ruby)                                |
-| Sucker Punch               | `sucker_punch`             | `>= 2.0`                 | `>= 2.0`                  | [Link][48]                          | [Link](https://github.com/brandonhilkert/sucker_punch)                       |
+- **Major version updates** (e.g. `1.0.0` to `2.0.0`) may change support for any runtime
+  from [Beta](#support-beta)/[GA](#support-ga)
+  to [Maintenance](#support-maintenance)/[Legacy](#support-legacy)/[EOL](#support-eol).
+- **Minor version updates** (e.g. `1.0.0` to `1.1.0`) may change support for any runtime
+  from [Beta](#support-beta)/[GA](#support-ga) to [Maintenance](#support-maintenance).
```

Comment:
I still think we're backing ourselves into a corner here. As a new Ruby comes out each year, this means that either we may need to carry a given Ruby version around for longer than we wanted to (sound familiar?) or we'll be forced to bump the major version sooner/when otherwise we wouldn't really need to.

Imagine in the extreme that we do no more features for ddtrace ever, only maintenance fixes. It's weird to think that a customer that onboards today on Ruby 3.3 will need to go through several "major" releases where actually nothing changes, because we just did a minor fix here or there, but we did them only because we wanted to drop support for older Rubies.

Furthermore, I don't really think this is a violation of semver, because support for this is built-into rubygems! Rubygems will not install a ddtrace version that doesn't work for a user's Ruby version, so if we drop support on a minor version (as many upstream libraries do), users would never accidentally pick up a version that wouldn't work for them.

Can we discuss this a bit before moving forward with this part?

Reply:
We discussed this privately via slack, so I'm happy with the current text :+1: